src/shared/cgroup-setup.c

```diff
@@ -152,19 +158,23 @@ bool cg_is_hybrid_wanted(void) {
 }
 
 bool cg_is_legacy_force_enabled(void) {
-        bool force;
+        static thread_local int enabled = -1;
+
+        /* If we have a cached value, return that. */
+        if (enabled >= 0)
+                return enabled;
 
         if (!cg_is_legacy_wanted())
-                return false;
+                return (enabled = false);
 
         /* If in container, we have to follow host's cgroup hierarchy. */
         if (detect_container() > 0)
-                return true;
+                return (enabled = true);
```

Comment:
This seems wrong. Previously `cg_is_legacy_wanted` is checked only after `cg_is_unified_wanted`, i.e. the former returns true as generic fallback. Now you're effectively enabling v1 for all containers since _unified_wanted calls _legacy_wanted first...